src/main/java/com/smartcar/sdk/data/BatchResponse.java

```diff
@@ -0,0 +1,101 @@
+package com.smartcar.sdk.data;
+
+import com.smartcar.sdk.SmartcarException;
+
+import org.joda.time.DateTime;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.GsonBuilder;
+import org.apache.commons.text.CaseUtils;
+import java.util.HashMap;
+import java.util.Iterator;
+
+public class BatchResponse extends ApiData {
+  private HashMap<String, JsonObject> data = new HashMap<>();
+
+
+  public  BatchResponse(String responsesArray) {
+
+    JsonParser parser = new JsonParser();
+    JsonElement responsesParsed = parser.parse(responsesArray);
+    JsonArray responsesJsonArray = responsesParsed.getAsJsonArray();
+    Iterator<JsonElement> responses = responsesJsonArray.iterator();
+
+    while (responses.hasNext()){
+      JsonObject res = responses.next().getAsJsonObject();
+      String path = res.get("path").getAsString();
+      this.data.put(path, res);
+    }
+  }
+
+  private static String toCamelCase(String fieldName) {
+    if (fieldName.contains("_")) { // checks for snake case
+      return CaseUtils.toCamelCase(fieldName, false, new char[]{'_'});
+    }
+    return fieldName;
+  }
+
+  static GsonBuilder gson = new GsonBuilder().setFieldNamingStrategy((field) -> toCamelCase(field.getName()));
+
+
+  public <T extends ApiData> SmartcarResponse<T> get(String path) throws SmartcarException {
+    JsonObject res = this.data.get(path);
+    Integer code = res.get("code").getAsInt();
+    if (code != 200) {
+      // TODO : handle error
+      System.out.println("ERROR");
```

Comment:
@sankethkatta do we want to throw one type of error here or many different kinds?

Reply:
I think we could wrap everything into the `SmartcarException` type.

Reply:
done!